repl/scala-2.11/src/main/scala/org/apache/spark/repl/SparkILoop.scala

```diff
@@ -44,7 +44,14 @@ class SparkILoop(in0: Option[BufferedReader], out: JPrintWriter)
     @transient val spark = if (org.apache.spark.repl.Main.sparkSession != null) {
         org.apache.spark.repl.Main.sparkSession
       } else {
-        org.apache.spark.repl.Main.createSparkSession()
+        try {
+          org.apache.spark.repl.Main.createSparkSession()
+        } catch {
+          case e: Exception =>
+            println("Failed to initialize Spark session:")
+            e.printStackTrace()
+            sys.exit(1)
```

Comment:
my concern is SparkILoop is used in a bunch of settings outside of spark and its shell/repl - sys.exit might not be ideal in some cases

Reply:
My usual response is "this is not a public class" (it's not in the public API docs), but let me see if it's easy to restrict the `sys.exit` to spark-shell invocations.